packages/cadl-ranch-specs/http/content-type/multipart-formdata/main.tsp

```diff
@@ -0,0 +1,74 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test content-type: multipart/form-data")
+@scenarioService("/content-type/multipart-formdata")
+namespace ContentType.MultipartFormData;
+
+@encode("binary")
+scalar binary extends bytes;
+
+model Address {
+  city: string;
+}
+
+model MultiPartRequest {
+  id: string; // Field "id" is text/plain
+  address: Address; // Field "address" is an application/json
+  profileImage: binary; // Field "profileImage" is application/octet-stream
+  previousAddresses: Address[]; // This is ONE part, as a JSON array
+  pictures: binary[]; // This is as many part as necessary
```

Comment:
Not sure if SDK really want to support this (assume it means any number of files).

Reply:
does openai not have an api like this?

Reply:
No. openai just got one audio file. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/OpenAI.Inference/models/audio/audio_transcription.tsp#L37-L42

Face API seems also one https://github.com/Azure/azure-rest-api-specs-pr/blob/fee4e6a2dc993c49e3c424848bd57e582cab23ca/specification/ai/Face/Sessions/models.tsp#L169-L170

I remember the other ask is 2 files.

Reply:
BTW, does `application/octet-stream` always require a filename? What if customer just want to upload a stream (not a file), do they need to set a fake one, or can ignore it?

Reply:
application/octet-stream as the request body has no filename informatin but when sent via multipart the part does have the filename info

Reply:
Yes, I just want to know if filename is *required* (I didn't dig deep into the RFC).

Reply:
If filename lacked, `multer` can't parse it

Reply:
Got it. Thanks.

Reply:
RFC (at least they say they are not get obsoleted by another RFC)

https://datatracker.ietf.org/doc/html/rfc7578#section-4.2

https://datatracker.ietf.org/doc/html/rfc2183#section-2.3